provisioning/miniprov/hare_mp/main.py

```diff
@@ -437,8 +437,8 @@ def start_mkfs_parallel(hostname: str, hare_config_dir: str):
         raise RuntimeError('One or more mkfs processes failed to start. '
                            'Please check the logs above for details.')
     else:
-        logging.info(f'Total time taken for all mkfs on this node = '
-                     f'{perf_result}\n\n')
+        logging.debug(f'Total time taken for all mkfs on this node = '
```

Comment:
If provisioner considers logging to the console as a sign of error, don't we need just to stop logging to console (and prefer files)?

Is this kind of contract documented btw? For me this restriction is a bit surprising..

Reply:
I was planning to create a separate jira for having the file writing function, and we can replace that function call at other places too.
I dont know if this was documented.
This was faced by @SwapnilGaonkar7 once and there was some discussion around this.
@SwapnilGaonkar7 , any idea on documentation of this?

Reply:
No idea about documentation, it was discussed in integration room
https://github.com/Seagate/cortx-prvsnr/pull/6209
Above PR is reference.
cc @mssawant

Reply:
From the above PR it seems that provisioner is erroring out if `rc !=0` or `stderr` is not empty. So are we adding anything to stderr?

Reply:
This is how they are returning the output, err and rc
``` def run(self, **args):
        """ This will can run simple process """
        for key, value in args.items():
            setattr(self, key, value)

        try:
            cmd = shlex.split(self._cmd) if isinstance(self._cmd, str) else self._cmd
            if getattr(self, 'realtime_output'):
                self.stdout=None
            self._cp = subprocess.run(cmd, stdout=self.stdout,
                    stderr=subprocess.PIPE, shell=self.shell, cwd=self.cwd,
                    timeout=self.timeout, env=self.env,
                    universal_newlines=self.universal_newlines)

            self._output = self._cp.stdout
            self._err = self._cp.stderr
            self._returncode = self._cp.returncode

        except TimeoutExpired as e:
            self._err = str(e)
            self._output = ''
            self._returncode = errno.ETIMEDOUT

        except CalledProcessError as e:
            self._err = str(e)
            self._output = ''
            self._returncode = e.returncode

        except Exception as err:
            self._err = "SubProcess Error: " + str(err)
            self._output = ''
            self._returncode = -1

        return self._output, self._err, self._returncode```

I wonder if they are checking `output` instead of `rc`, cc @knekrasov , @suvratjoshi , @vaibhavparatwar

Reply:
Console logger uses `stderr`. 
```python
def setup_logging() -> None:
    console = logging.StreamHandler(stream=sys.stderr)
    fhandler = logging.handlers.RotatingFileHandler(LOG_FILE,
                                                    maxBytes=LOG_FILE_SIZE,
                                                    mode='a',
                                                    backupCount=5,
                                                    encoding=None,
                                                    delay=False)
    logging.basicConfig(level=logging.INFO,
                        handlers=[console, fhandler],
                        format='%(asctime)s [%(levelname)s] %(message)s')

```

Reply:
Sorry, looks like utils code is reading the arguments in correct order.

Reply:
So are we posting the log message in stderr? or returning a non-zero rc value in this case?

Reply:
>Console logger uses stderr.
Ah, @knekrasov , @suvratjoshi , then we need to fix that, no?

Reply:
We can @mssawant :-)
Up to this point it wasn't a problem that somebody would need to fix. But if the requirements have changed - then yes, let's adjust our logic.

Reply:
Updated!

Reply:
>let's adjust our logic.

Not really about adjusting but in general regular logging shouldn't go to stderr anyway, right.

Reply:
It really depends. Sometimes the console application prints something and that printing is an actual result of its work. But additionally it may also print something like progress percents, debug messages about connections etc. In that case it may be helpful to redirect stderr to `/dev/null` or to a file to separate those outputs.

For instance, even our `cfgen` may print something when it is run with `--dry-run` flag. So it would be helpful to allow caller to distinguish the output produced by the logger and valid printing.

Long story short, I don't think that there is a definite rule "printing to stderr is allowed if and only if there is a problem in your app". 

Example: `wget` tool prints the progress information to `stderr`
```
wget --progress=dot -O - http://google.com 2>./wgetout
<contents of the http://google.com webpage is printed>

cat ./wgetout
--2021-12-03 08:51:53--  http://google.com/
Resolving google.com (google.com)... 142.250.190.110, 2607:f8b0:4009:80b::200e
Connecting to google.com (google.com)|142.250.190.110|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: http://www.google.com/ [following]
--2021-12-03 08:51:53--  http://www.google.com/
Resolving www.google.com (www.google.com)... 142.250.190.4, 2607:f8b0:4009:808::2004
Connecting to www.google.com (www.google.com)|142.250.190.4|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘STDOUT’

     0K .......... .....                                        673K=0.02s

2021-12-03 08:51:53 (673 KB/s) - written to stdout [16029]
```